python/pyarrow/_s3fs.pyx

```diff
@@ -312,3 +328,25 @@ cdef class S3FileSystem(FileSystem):
         The AWS region this filesystem connects to.
         """
         return frombytes(self.s3fs.region())
+
+    @property
+    def allow_bucket_creation(self):
+        """
+        Whether to allow CreateDir at the bucket-level.
```

Comment:
We should probably use the same docstring for constructor parameters and the associated properties.
Personally, I like "Whether to allow CreateDir at the bucket-level."